redis/tests/support/mod.rs

```diff
@@ -39,12 +39,64 @@ pub fn current_thread_runtime() -> tokio::runtime::Runtime {
     builder.build().unwrap()
 }
 
-pub fn block_on_all<F>(f: F) -> F::Output
+#[cfg(feature = "aio")]
+pub fn block_on_all<F, V>(f: F) -> F::Output
 where
-    F: Future,
+    F: Future<Output = redis::RedisResult<V>>,
 {
-    current_thread_runtime().block_on(f)
+    use std::panic;
+    use std::sync::atomic::{AtomicBool, Ordering};
+
+    static CHECK: AtomicBool = AtomicBool::new(false);
+
+    // TODO - this solution is purely single threaded, and won't work on multiple threads at the same time.
+    // This is needed because Tokio's Runtime silently ignores panics - https://users.rust-lang.org/t/tokio-runtime-what-happens-when-a-thread-panics/95819
+    // Once Tokio stabilizes the `unhandled_panic` field on the runtime builder, it should be used instead.
+    panic::set_hook(Box::new(|panic| {
+        println!("Panic: {panic}");
+        CHECK.store(true, Ordering::Relaxed);
+    }));
+
+    // This continuously query the flag, in order to abort ASAP after a panic.
+    let check_future = futures_util::FutureExt::fuse(async {
+        loop {
+            if CHECK.load(Ordering::Relaxed) {
+                return Err((redis::ErrorKind::IoError, "panic was caught").into());
+            }
+            futures_time::task::sleep(futures_time::time::Duration::from_millis(1)).await;
```

Comment:
why such a small interval?

Reply:
we want to be able to catch any small await point, so we need to check often.